### PR TITLE
Feature KHPW10 Create own user model

### PR DESCRIPTION
**Pull request by SofiiaOmelchenko**

### CODE REVIEW

File: happykh/users/models.py [hunk 1]
```diff
@@ -1,3 +1,48 @@
...
+
+class User(AbstractBaseUser):
+    """"
+    Customized user model with email as username and additional fields
+    """
+    email = models.EmailField(max_length=255, unique=True)
+    first_name = models.CharField(max_length=30, blank=True)
+    last_name = models.CharField(max_length=30, blank=True)
+    age = models.PositiveSmallIntegerField()
+    gender = models.CharField(choices=(('Ж', 'woman'), ('М', 'man')), max_length=2, default='Ж')
```
Comment (Formeme) [inline happykh/users/models.py]: Вынести кортеж выборов в отдельный кортеж и "рабочие названия" каждой пары записать через переменные.

File: happykh/users/models.py [hunk 2]
```diff
@@ -38,11 +38,26 @@ class User(AbstractBaseUser):
...
     gender = models.CharField(choices=(('Ж', 'woman'), ('М', 'man')), max_length=2, default='Ж')
     profile_image = models.ImageField(null=True, blank=True)
+    is_active = models.BooleanField(default=True)
     is_admin = models.BooleanField(default=False)
     objects = UserManager()
 
     USERNAME_FIELD = 'email'
     REQUIRED_FIELDS = []
+
+    def has_perm(self, perm, obj=None):
```
Comment (Formeme) [inline happykh/users/models.py]: I think it is a bad idea to rewrite built-in function this way(with static return). I think it is better to use PermissionsMixin.

File: happykh/users/models.py [hunk 3]
```diff
@@ -38,11 +38,26 @@ class User(AbstractBaseUser):
...
+        "Does the user have a specific permission?"
+        return True
+
+    def has_module_perms(self, app_label):
```
Comment (Formeme) [inline happykh/users/models.py]: I think it is bad to have an unused function in a code.

File: happykh/users/admin.py [hunk 4]
```diff
@@ -1,3 +1,4 @@
+"""" """
```
Comment (Formeme) [inline happykh/users/admin.py]: I think it is bad to have blank docstring.

File: happykh/users/admin.py [hunk 5]
```diff
@@ -10,12 +11,12 @@ class UserAdmin(BaseUserAdmin):
     # These override the definitions on the base UserAdmin
     # that reference specific fields on auth.User.
     list_display = ('email', 'is_admin')
-    list_filter = ('is_admin',)
+    list_filter = ()
```
Comment (Formeme) [inline happykh/users/admin.py]: I think it is bad idea to have blank variable in a code.

File: happykh/users/admin.py [hunk 6]
```diff
@@ -1,3 +1,78 @@
...
+            'classes': ('wide',),
+            'fields': ('email', 'age', 'password1', 'password2')}
+         ),
+    )
+    search_fields = ('email',)
+    ordering = ('email',)
+    filter_horizontal = ()
+
+
+admin.site.register(User, UserAdmin)
```
[the remark this answers is not in the document]
Comment (SofiiaOmelchenko) [reply happykh/users/admin.py]: Maybe GitHub doesn't add the last line, but it exists in PyCharm.

**Pull request closed without merging**